### ISSUE TITLE
Followers/follows failed to migrate on move to new instance.

### ISSUE THREAD

**Issue by AdrianEdelen** [mastodon/mastodon#25773]
### Steps to reproduce the problem

1. Create account on new instance
2. Create alias on new instance
3. redirect account
...

### Expected behaviour

follows and followers transfer to new new account

### Actual behaviour

no follows or followers transfered 

### Detailed description

I performed the standard procedure for transferring to a different instance. and the actual redirect and migration worked. However when selecting the button to initiate the migration, an error was shown "something went wrong" and the followers/follows did not transfer. it is not a big deal for me but wanted to memorialize the issue. I was able to manually import my follows list, but of course not followers.

### Specifications

Unsure of the version on the 'from' instance, whatever version mastodon.social was on yesterday
for the 'to' instance it is mastodon v4.1.2

**Comment by ClearlyClaire** [comment]
> redirect account

Account redirections and migrations are not the same feature. Are you sure you actually performed a migration? (Reading the remaining of the issue, it appears to be the case, but I'd like to double-check)

> However when selecting the button to initiate the migration, an error was shown "something went wrong" and the followers/follows did not transfer.

Ok, that does sound like an issue with the migration code. I will see with mastodon.social devops if there are logs, or otherwise try to replicate the issue.

**Comment by ClearlyClaire** [comment]
I couldn't find the error nor could I reproduce it :/

**Comment by AdrianEdelen** [comment]
I definitely did a proper migration as well as a redirect, sorry I phrased it poorly initially. I am not too concerned about it, I just wasn't sure if this was a common problem. 
for more information:
Here I can confirm that the migration did work: 
![Capture](https://github.com/mastodon/mastodon/assets/6466981/1c6c4c74-d430-4f45-8a0a-ea73f8134747)
and here we can see that the old account still has those followers:
![Capture](https://github.com/mastodon/mastodon/assets/6466981/435b2edc-367a-46fc-baa2-1acb2d20ec09)

is the migration process typically reliable?
as far as I understand, the followers should have been removed from the old account. is it possible that an issue on the 'to' instance could prevent the migration from happening?

**Comment by RonaldPK** [comment]
The same happened to me, on 5 July. I also tried to migrate from mastodon.social, got the error page, and no followers were transferred and no redirect was set up. So I ended up with the same screen as Adrian. 

To me it's quite annoying since I have to wait 30 days (26 by now) before I can try again.

**Comment by marcus-herrmann** [comment]
I can only add into the choir, I started my migration from mastodon.social to a Mastohost instance on the evening of the July 7th. Until now, only 41 of 615 followers have moved, with ~10h between one and the next.

Also got the "something went wrong message" at one point

**Comment by marcus-herrmann** [comment]
And 30 days later, still only ~60 accounts migrated and I can't restart the migration, because, according to the interface, I already moved that account. And Mastodon.social is not replying to multiple e-mails regarding this issue.

**Comment by ClearlyClaire** [comment]
> as far as I understand, the followers should have been removed from the old account. is it possible that an issue on the 'to' instance could prevent the migration from happening?

What happens (at least what should happen) is the old server tells all your followers you have migrated to a new account. When receiving such information, Mastodon fetches the new account, checks that it's indeed marked as a new account for your old account, issue a follow request, and then unfollows the old account.

If the destination server is inaccessible during the process, your followers won't be able to fetch the new account and issue a follow request. This usually does not matter much, as the process is retried a few times over a few days.

Of course, a bug could happen on the old server, on the new server, or on the followers' servers, though no such bug is currently identified.

> The same happened to me, on 5 July. I also tried to migrate from mastodon.social, got the error page, and no followers were transferred and no redirect was set up. So I ended up with the same screen as Adrian.

This sounds slightly different than what was initially reported, which included the redirect being set up and some followers being migrated.

> Also got the "something went wrong message" at one point.

do you remember at which point that was?

> And 30 days later, still only ~60 accounts migrated and I can't restart the migration, because, according to the interface, I already moved that account.

You should be able to undo the move and do it again.

**Comment by marcus-herrmann** [comment]
> do you remember at which point that was?

No, I ddn't remember 30 days ago, otherwise I would have written it down. Now a month later it did not suddenly appear in my memory.

> You should be able to undo the move and do it again.
How?

But you don't need to answer. I stopped on the own-instance endeavor, and will be staying at mastodon.social. Grumpy.

**Comment by ClearlyClaire** [comment]
> How?

Replying for completeness, if someone stumbles on this issue: there should be a “Cancel redirect” link.

![image](https://github.com/mastodon/mastodon/assets/384364/a4a02247-9f1f-412b-bb83-1c969f9703cd)

> But you don't need to answer. I stopped on the own-instance endeavor, and will be staying at mastodon.social. Grumpy.

Sorry we could not figure out what was going on and help you with it.

**Comment by marcus-herrmann** [comment]
> Replying for completeness, if someone stumbles on this issue: there should be a “Cancel redirect” link.

Oh, I did this. Otherwise I could not have reactivated my .social account. No option to undo.

**Comment by thissophie** [comment]
My experience of this - migrating to a new account on the same instance  (v4.4.5) - was only 25% of followers actually moved

**Comment by ClearlyClaire** [comment]
Are you comfortable with sharing information about your account and followers? If so, we could try to see if we can see a pattern in the followers which did not follow the new account.

**Comment by marcus-herrmann** [comment]
To chime in once again: Last time I moved instances, only the mastodon.social ones moved (I moved from mastodon.social - maybe that's a pattern for you). And I triggered the move several times, to no avail.

**Comment by thissophie** [comment]
> Are you comfortable with sharing information about your account and followers? If so, we could try to see if we can see a pattern in the followers which did not follow the new account.

From https://cloudisland.nz/@patrick to https://cloudisland.nz/@thisisme 

Like others it looks like the ones that were migrated were from my same instance (but not _all_ from my instance were migrated, as I think some followed me manually after I started following people from the new account)

**Comment by ClearlyClaire** [comment]
Thank you! It indeed looks like local followers updated promptly, but none, or almost none of the remote ones… I'll see if we can work with your instance admin to identify an issue, thanks!

**Comment by gsaich** [comment]
FYI - same problem.  I followed the migration steps because my current server is shutting down.  I had 1099 followers and as of today I only have 185 on my new server - I believe most of them are people who saw an announcement from me and re-followed me.  I tried to re-run the Move to a different account / Move followers routine and received the error:  Something isn't quite right yet! Please review the error below -  [xxxxxxxx] is the same account you have already moved to.  So it looks like 914 followers did not migrate.  Any idea how to recover this.  I did not delete my old account, the server is up until October - it still shows I have 1099 followers there, and show the message saying i have moved to the new account.  Can this be fixed?  Any ideas?

**Comment by gunchleoc** [comment]
Here's another example of a failed migration, looks like they ran into timeouts - some of the followers were set to Follow Request status, so they had to unfollow and hit the follow button again https://fnordon.de/@OskarImKeller/117195609735692483
